src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
+            head = tail = newNode;
+            size++;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(value);
+        if (index < 0 || index > size()) {
+            throw new IndexOutOfBoundsException("Index " + index
+                    + " must be less than " + size());
+        }
+        if (head == null) {
+            head = tail = newNode;
+            size++;
+        } else if (index == 0) {
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+            size++;
+        } else if (index == size()) {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+            size++;
+        } else {
+            Node<T> currentNode = findElementByIndex(index);
+            newNode.next = currentNode;
+            newNode.prev = currentNode.prev;
+            currentNode.prev.next = newNode;
+            currentNode.prev = newNode;
```

Comment:
let's try to simplify this
first, reuse add(value)
we should call it index == size
```
        if (index == size) {
            add(value);
            return;
        }
```
then get node by index, create node to be inserted
You will need a constructor for
```
        Node<T> node = findElementByIndex(index);
        Node<T> newNode = new Node<>(node.prev, value, node);
```
or 
```
        Node<T> newNode = new Node<>(value);
        newNode.prev = node.prev;
        newNode.next = node;
```
and we can do everything else with just one neat `if-else`
```
        if (newNode.prev == null) {
            head = newNode;
        } else {
            newNode.prev.next = newNode;
        }

        node.prev = newNode;
        size++;
```